### ISSUE TITLE
Infinite loop

### ISSUE THREAD

**Issue by komlev** [travco/postcss-extend#12]
This code sends plugin into infinite loop
```css
.button {
  all: initial;
  cursor: pointer;
  font-size: 16px;
  padding: 10px 16px;
  border-radius: 2px;
}

.button_state_active {}

.button_state_hover {}

.button_state_focus {}

.button_state_disabled {
  cursor: default !important;
}

.button:hover:not(.button_state_disabled) {
  @extend .button_state_hover;
}

.button:focus:not(.button_state_disabled) {
  @extend .button_state_focus;
}

.button:active:not(.button_state_disabled) {
  @extend .button_state_active;
}

.button_state_hover {
  color: #FFF;
}

.button_state_active {
  color: #FFF;
}

.button_state_focus {

  @extend .button_state_hover;
}

.button_state_disabled {
  background-color: disabled(#FFF);
}

.button_size_smaller {
  padding: 4px 16px;
}
```

**Comment by ai** [comment]
@travco do you need any help here?

**Comment by travco** [comment]
No, I think I found it @ai. Appreciate the offer.

Turns out this was a case that was omitted in the test-set because it wasn't a worst-case scenario... it was simple. Unfortunately it's also simple enough that the recursion-handler I built has problems progressing past it, keeps slipping back expecting to be able to extend it further. Added a quick statement to delete on a resolved look-ahead, did not add a statement to remove the palm-shaped mark on my face.

@komlev Most recent commit on the repo should resolve the problem, I'm holding off on an NPM release today until I fix the other glaring issues.
